lib/Common/Memory/RecyclerWriteBarrierManager.cpp

```diff
@@ -30,6 +33,7 @@ BYTE* RecyclerWriteBarrierManager::cardTable = RecyclerWriteBarrierManager::x64C
 #else
 // Each byte in the card table covers 4096 bytes so the range covered by the table is 4GB
 BYTE RecyclerWriteBarrierManager::cardTable[1 * 1024 * 1024];
+bool dummy = RecyclerWriteBarrierManager::Initialize();
```

Comment:
because of the initialization order, before the card table is allocated, we are already allocating dictionary etc with heap allocator. and in "struct _ArrayWriteBarrier" RecyclerWriteBarrierManager is not availble yet, otherwise we need to change the include and initialization order
the g_ function should only be called when the flag is enabled, I'll fix that part.

---
In reply to: [86643214](https://github.com/Microsoft/ChakraCore/pull/1920#discussion_r86643214) [](ancestors = 86643214,86642266)